### DISCUSSION TITLE
argo-workflows/client-libraries/

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#9537] [General]
# argo-workflows/client-libraries/

https://argoproj.github.io/argo-workflows/client-libraries/

**Comment by liuyix** [comment]
It's a very painful fact that java client library is not 'maven-native'. It costs me hours and I still fail to configure correctly. :-(

**Reply by aklink** [reply]
The python client library works fine and is easy to use. Althought it's a pitty to have to call a python library out of java, just because the native java client library is not usable straight-forward. When it's easier to use the python client library out of java than using the native java client library, then something is wrong with usability of the java client library.
By the way: Argo Workflows is great! Despite of the lack of usability of the java client library.
